### ISSUE TITLE
replace moment-timezone dependency with spacetime

### ISSUE THREAD

**Issue by leeoniya** [grafana/grafana#28922]
**What would you like to be added**:

replace `moment-timezone` dependency with `spacetime`:

http://spacetime.how/
https://github.com/spencermountain/spacetime

**Why is this needed**:

467.6 kB:
https://bundlephobia.com/result?p=moment-timezone@0.5.31

47.4 kB:
https://bundlephobia.com/result?p=spacetime@6.9.0

**Comment by torkelo** [comment]
Like the idea, is it possible without first replacing moment?

**Comment by leeoniya** [comment]
> Like the idea, is it possible without first replacing moment?

i havent looked too much into this, but i'm sure it depends on exactly how and where moment is used in the grafana codebase.

**Comment by mckn** [comment]
Sweet, this could be an alternative as well: https://date-fns.org/

**Comment by leeoniya** [comment]
@mckn i think that's a good alternative to moment but it still needs the huge tz database for timezone support: https://bundlephobia.com/result?p=date-fns-timezone@0.1.4

**Comment by mckn** [comment]
> @mckn i think that's a good alternative to moment but it still needs the huge tz database for timezone support: https://bundlephobia.com/result?p=date-fns-timezone@0.1.4

Yes, at the same time it looks like the `spacetime` is missing parts of the dataset such as abbreviations and country mapping. But I like the API so maybe we can add those data points in some other way?

**Comment by leeoniya** [comment]
it looks like grafana exposes momen't full/extensive formatting API, which might make this impossible :(

https://grafana.com/docs/grafana/latest/administration/configuration/#date_formats
